zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -84,6 +87,20 @@ public ZkClientUriDomainMappingHelper(ZooKeeperServer zks) {
     parseZNodeMapping();
   }
 
+  /**
+   * @return True if the new updater is setup to the helper instance. False if the specified updater is not set since
+   * another updater has already been configured.
+   */
+  synchronized boolean setDomainAuthUpdater(ConnectionAuthInfoUpdater updater) {
```

Comment:
What is the intent of returning a boolean? Since the return value isn't really used anywhere.

Should we just do a lazy initialization? Also upon giving a little more thought, this part of the logic is single-threaded, so I'm not sure if we need to make it thread-safe necessarily.

Reply:
Is it because you synchronize on the updater object in `updateDomainBasedAuthInfo` method?  We can synchronize on a final object to save the troubles.